File: hw/gudino/u3/hw28PrinciplesAplication/ComputerStore/productos_app/models/tax_calculator.py
from abc import ABC, abstractmethod

class TaxCalculator(ABC):
    @abstractmethod
    def calculate_final_price(self, base_price: float) -> float:
        pass

class SimpleTaxCalculator(TaxCalculator):
    TAX_RATE = 0.15
    
    def calculate_final_price(self, base_price: float) -> float:
        return base_price + (base_price * self.TAX_RATE)

File: hw/gudino/u3/hw28PrinciplesAplication/ComputerStore/compujs/models/TaxCalculator.js
class TaxCalculator {
    calculateFinalPrice(basePrice) {
        throw new Error("Not implemented");
    }
}

class SimpleTaxCalculator extends TaxCalculator {
    static TAX_RATE = 0.15;

    calculateFinalPrice(basePrice) {
        return basePrice + basePrice * SimpleTaxCalculator.TAX_RATE;
    }
}

module.exports = { TaxCalculator, SimpleTaxCalculator };
